[Docs]: add info about lower scope tokens for gitlab (#9017)
All-Hands-AI/OpenHands: docs/usage/cloud/gitlab-installation.mdx

![Connect Repo](/static/img/connect-repo.png)

## Using Tokens with Reduced Scopes

> OpenHands requests an API-scoped token during OAuth authentication. By default, this token is provided to the agent.
> To restrict the agent's permissions, you can define a custom secret `GITLAB_TOKEN`, which will override the default token assigned to the agent.
> While the high-permission API token is still requested and used for other components of the application (e.g. opening merge requests), the agent will not have access to it.

Reviewer: ```suggestion
OpenHands requests an API-scoped token during OAuth authentication which is used by default.
To restrict OpenHands' permissions, you can define a custom secret `GITLAB_TOKEN`, which will override the default token used.
While the high-permission API token is still requested and used for other processes (e.g. opening merge requests), OpenHands will not have access to it.
```

Reviewer: @rbren in this scenario, how would you distinguish them?

Reviewer: I can certainly see the confusion.

Reviewer 2: Yeah that's an interesting distinction. I do think saying "the agent" to contrast that from the app will be important in some scenarios
